src/Microsoft.DotNet.Build.Tasks/DownloadFileFromUrl.cs

```diff
@@ -0,0 +1,92 @@
+using Microsoft.Build.Framework;
+using System;
+using System.IO;
+using System.Net.Http;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class DownloadFileFromUrl : BuildTask
+    {
+        
+        /// <summary>
+        /// The url to download the file from.
+        /// </summary>
+        [Required]
+        public string DownloadSource { get; set;}
+        
+        /// <summary>
+        /// The directory where the file will be downloaded to.
+        /// </summary>
+        [Required]
+        public string DestinationDirectory { get; set; }
+
+        /// <summary>
+        /// The file where the source will be downloaded to.
+        /// </summary>
+        [Required]
+        public string DestinationFileName { get; set; }
+
+        /// <summary>
+        /// The default is to emit warnings and not fail, the caller can fail if output parameter DownloadedFile is empty.
+        /// Set this to true to fail instead of just warn.
+        /// </summary>
+        public bool FailOnError { get; set; }
+
+        /// <summary>
+        /// The file that was downloaded, if there is an error this will be empty.
+        /// </summary>
+        [Output]
+        public string DownloadedFile { get; set; }
```

Comment:
If we combine the two "destination" input parameters into "DestinationPath" (for example), then this will just be equivalent to that.